test/modules/http/regressions/http-socket-enetunreach.test.ts

```diff
@@ -0,0 +1,14 @@
+import { it, expect } from 'vitest'
+import { httpGet } from '../../../helpers'
+import { ClientRequestInterceptor } from '../../../../src/interceptors/ClientRequest'
+
+it('does not leave the test process hanging due to the custom socket timeout', async () => {
+  const interceptor = new ClientRequestInterceptor()
+  interceptor.apply();
```

Comment:
May I please ask you to move the `ClientRequestInterceptor` instance outside of the test and enable/dispose of it in the `beforeAll`/`afterAll` hooks imported from `vitest`? Resource cleanup is important 😃

Reply:
Fixed